app/models/ability.rb

```diff
@@ -273,6 +275,7 @@ def taxonomies
     can [:index, :show], Permission
     can [:index, :show], UserRole
     can [:index, :show], CancellationReason
+    can [:index, :show], Organisation
```

Comment:
This is also a problem. It is allowing any user to view or list any organisation details. Will need to secure this to only organisations that the user belongs to.

Reply:
Hi @steveyken 

On browse we have new feature where if user do not exist in system then we are directly creating user with mobile and sending pin for login. After that we redirect user to account details page where we have select list for organisations and yet user is not authenticated as per role or permission in our system. Even in this case we want to show list of organisation's to user to select organisation.

That's the case where we kept it open. Hope that makes sense. Correct me if I am wrong. 

Thanks.

Reply:
Yes good point. However, we should make sure that it's not all the organisation details that are exposed. Just the names in the drop-down list. The permissions above are enabling access to all details.

Reply:
One solution here would be to create a `api/v1/organisations/names` endpoint. This could use an OrganisationNamesSerializer that returns just the `{id: name}` (translated of course). The ability.rb file could be adjusted so that anonymous users can access `api/v1/organisations/names.json` but NOT `api/v1/organisations.json` or `api/v1/organisations/1.json` This would leave the 'show' action available for users who are part of the organisation. Whilst it might not make a lot of difference right now, we will implemented access control shortly and it's better to have this change in now rather than forget it later.

What do you think?